src/tiled/mainwindow.cpp

```diff
@@ -569,6 +571,53 @@ MainWindow::MainWindow(QWidget *parent, Qt::WindowFlags flags)
             });
         }
     });
+    connect(mUi->actionNewWorld, &QAction::triggered, this, [this,preferences]{
+        QString lastPath = preferences->lastPath(Preferences::WorldFile);
+        QString filter = tr("All Files (*);;");
+        QString worldFilesFilter = tr("World files (*.world)");
+        filter.append(worldFilesFilter);
+
+        MapEditor* mapEditor = static_cast<MapEditor*>(DocumentManager::instance()->editor(Document::DocumentType::MapDocumentType));
+        QString worldFile = QFileDialog::getSaveFileName(mapEditor->editorWidget(), tr("New Map"), lastPath,
+                                                         filter, &worldFilesFilter);
+        if (worldFile.isEmpty() || QFile::exists(worldFile))
+            return;
+
+        preferences->setLastPath(Preferences::WorldFile, QFileInfo(worldFile).path());
+        QString errorString;
+        if (!WorldManager::instance().addEmptyWorld(worldFile, &errorString)) {
+            QMessageBox::critical(this, tr("Error Creataing World"), errorString);
+        } else {
+            const auto worldFiles = WorldManager::instance().loadedWorldFiles();
+            mSettings.setValue(QLatin1String("LoadedWorlds"), QVariant(worldFiles));
+            mUi->menuUnloadWorld->setEnabled(!worldFiles.isEmpty());
+        }
+    });
+    connect(mUi->menuSaveWorld, &QMenu::aboutToShow, this, [this] {
```

Comment:
I think you're right, although here it doesn't appear to be disabled, regardless of whether a world is loaded or not. Do you think it's a big deal if we just keep the menu itself active even when there's no items in it?

Reply:
I think disabling when there are no worlds to save is good, so a possible solution would be to not wait with populating the submenu until it is about to show. However, if we decide to move towards a world editor instead of a tool then there is no point in fixing this behavior since this menu item will go away.

Reply:
Your call, I'll not touch it until you decide. ( same for the usability improvements )